File: service/src/main/java/com/bnorm/barkeep/service/RecipeService.java
// Copyright 2016 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.service;

import java.util.Collection;

import com.bnorm.barkeep.model.Recipe;

public interface RecipeService {

  Collection<Recipe> listRecipes();

  Recipe getRecipe(long recipeId);

  Recipe createRecipe(Recipe recipe);

  Recipe setRecipe(long recipeId, Recipe recipe);

  void deleteRecipe(long recipeId);
}

File: service/src/main/kotlin/com/bnorm/barkeep/service/RecipeService.kt
// Copyright 2017 (C) BNORM Software. All rights reserved.
package com.bnorm.barkeep.service

import com.bnorm.barkeep.model.Recipe
import com.bnorm.barkeep.model.RecipeSpec

interface RecipeService {

  fun getRecipes(): Collection<Recipe>

  fun getRecipe(id: Long): Recipe?

  fun createRecipe(recipe: RecipeSpec): Recipe

  fun setRecipe(id: Long, recipe: RecipeSpec): Recipe

  fun deleteRecipe(id: Long)
}
